light_group_mixer/light_group_mixer.py

```diff
@@ -0,0 +1,244 @@
+import nuke
+
+
+def update_light_group_mixer() -> None:
+    """This function adds/removes our lightgroups to/from our gizmo."""
+    lightgroup_mixer_node = nuke.thisNode()
+
+    channels = _get_all_channels(lightgroup_mixer_node)
+    if channels is None:
+        return
+
+    all_lightgroups, lightgroups_to_build = _get_lightgroups(
+        lightgroup_mixer_node, channels
+    )
+
+    to_connect_to = _get_node_to_connect_to()
+
+    _add_lightgroups(lightgroups_to_build, to_connect_to, lightgroup_mixer_node)
+
+    lightgroups_to_remove = _get_old_unused_lightgroups(
+        lightgroup_mixer_node, all_lightgroups
+    )
+    _remove_lightgroupgrade_nodes(lightgroup_mixer_node, lightgroups_to_remove)
+
+    reset_solos()
+
+
+def _get_all_channels(lightgroup_mixer_node: nuke.Node) -> list:
+    """This function returns all the channels from the input node."""
+    try:
+        return lightgroup_mixer_node.input(0).channels()
+    except AttributeError:
+        nuke.message("You haven't connected your input node.")
+        return None
+
+
+def _get_node_to_connect_to() -> nuke.Node:
+    """This functions returns the node our next node should connect to."""
+    try:
+        if nuke.toNode("Output1").input(0).Class() == "LightGroupGrade":
+            to_connect_to = nuke.toNode("Output1").input(0)
+        else:
+            to_connect_to = nuke.toNode("Input1")
+
+    except AttributeError:
+        to_connect_to = nuke.toNode("Input1")
+
+    return to_connect_to
+
+
+def _get_lightgroups(lightgroup_mixer_node: nuke.Node, channels) -> tuple:
+    """This function returns all the lightgroups and the lightgroups we still need to build."""
+    all_knobs = lightgroup_mixer_node.knobs()
+    all_lightgroups = []
+    lightgroups_to_build = []
+    for channel in channels:
+        if not channel.startswith("LG_"):
+            continue
+
+        # Seeing as lightgroups contains several subchannels, we just
+        # pick one. Blue is my favorite color, so I went with that :)
+        if not channel.endswith("blue"):
+            continue
+
+        all_lightgroups.append(channel.split(".")[0])
+
+        if f"{channel.split('.')[0]}_tab_begin" in all_knobs:
+            continue
+
+        lightgroups_to_build.append(channel)
+
+    return all_lightgroups, lightgroups_to_build
+
+
+def _add_lightgroups(
+    lightgroups_to_build, to_connect_to, lightgroup_mixer_node: nuke.Node
```

Comment:
Would be good to rename the arguments. For example to_connect_to doesn't really describe what it is. Probably node_to_connect_to describes it better.

Also add some typehints to the arguments